src/Logging/TeamCity.php

```diff
@@ -27,37 +30,53 @@ final class TeamCity extends DefaultResultPrinter
     private const TEST_SUITE_STARTED  = 'testSuiteStarted';
     private const TEST_SUITE_FINISHED = 'testSuiteFinished';
 
+    /**
+     * A stack of error messages and test failures to be displayed
+     * once the test suite has finished running.
+     *
+     * @var array<callable>
+     */
+    protected $outputStack = [];
+
     /** @var int */
     private $flowId;
 
     /** @var bool */
     private $isSummaryTestCountPrinted = false;
 
-    /** @var \PHPUnit\Util\Log\TeamCity */
-    private $phpunitTeamCity;
+    /** @var \NunoMaduro\Collision\Adapters\Phpunit\Printer|null Printer */
+    private $collisionPrinter = null;
 
-    public function __construct(bool $verbose, string $colors)
+    /**
+     * @param resource|string|null $out
+     */
+    public function __construct($out, bool $verbose, string $colors)
     {
-        parent::__construct(null, $verbose, $colors, false, 80, false);
-        $this->phpunitTeamCity = new \PHPUnit\Util\Log\TeamCity(
```

Comment:
I might be a bit blind, but what replaced the part which did this? 🤔

Reply:
Basically nothing. We're firing all of the events that team city were firing manually, and now that we don't share their output at all, it became redundant. Can you see an issue with it?

Reply:
I've gone through and manually implemented some of the events. We have to do it manually so as not to give PHPStorm too much information (because then it writes to the console) but just enough to get it to display the correct icon. If we decorate the base `TeamCity` logger it writes out too much to the console.